tuples.md

```diff
@@ -0,0 +1,67 @@
+# Writing Assignment - Tuples
+
+So far you have seen how to use arrays for holding a set of values and then iterate through them, for example:
+
+```swift
+var distances: [Int] = [10, 15, 20]
+
+for distance in distances {
+  print(distance)
+}
+```
+
+Arrays are a useful method for collecting variables, but limited as the values they contain can only be of one type. In this exercise we will introduce another, more flexible content type, the tuple.
+
+A tuple is a flexible variable type that allows you to group multiple variables into one convenient compound variable. The variables contained within a tuple can be of any type (for example a string, integer or even arrays) and you can mix the types. This makes tuples useful for creating a container variable for related values, for example:
+
+```swift
+let completedDistance = ("km", 10, true)
+```
+
+This tuple represents an entry from a fitness tracker, recording a distance unit as a string, a distance as an integer and a boolean to show completion.
+
+**Note**: Tuples are designed for short term groups of related data and not for data that is accessed frequently in an application. Long term data storage is generally handled by classes, which you will learn later.
+
+You can access the values inside a tuple using index numbers, for example:
+
+```swift
+print ("You ran \(completedDistance.1)\(completedDistance.0)")
+```
+
+But a better method is to decompose the tuple structure into separate variables and access those variables, for example:
+
+```swift
+let (distanceUnit, distanceQuantity, distanceAchieved) = completedDistance
+print ("You ran \(distanceQuantity)\(distanceUnit)")
+```
+
+This method also lets you tidily mix the values in the tuple with text descriptions, or more practically, the labels of interface elements.
+
+A perfect use of tuples is to take their value(s) from a function, for example:
+
+```swift
+func formatRun(userName: String) -> (distanceUnit: String, distanceQuantity: Int, distanceAchieved: Bool) {
```

Comment:
Using examples is great! One thing we always try to do is make it very readable because our audience is at beginner level. Even though in practice, they might not create a tuple that represents a movie (as it would make sense to turn it into a class/struct), I like the idea of a function where the return type is a tuple (like you did here) but having the names of the individual elements be more distinct to avoid any confusion with a beginner.

At this point, the student will also not be super secure with functions. When they see that the userName argument isn't being used, they might question that and then question whether or not they fully grasp functions. So, I would prefer then for this function to look like this. Creating comments to let them know work would normally be done here to get the info is more clear.

```
func formatRun(userName: String) -> (distanceUnit: String, distanceQuantity: Int, distanceAchieved: Bool) {
    
    // Do some work with the userName parameter to get the necessary info used below (km, 100 and true)
    
    let unit = "km"
    let quantity = 100
    let achieved = true
    
    return (unit, quantity, achieved)
}
```